src/crewai/utilities/events/listeners/tracing/types.py

```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass, field, asdict
+from datetime import datetime, timezone
+from typing import Dict, Any
+import uuid
+
+
+@dataclass
+class TraceEvent:
+    """Individual trace event payload"""
+
+    event_id: str = field(default_factory=lambda: str(uuid.uuid4()))
+    timestamp: str = field(
+        default_factory=lambda: datetime.now(timezone.utc).isoformat() + "Z"
```

Comment:
Avoid appending "Z" manually.. `datetime.now(timezone.utc).isoformat()` already includes timezone info as "+00:00" (UTC timezone)

adding manually "Z" may result in misleading or malformed timestamps like the following example

```
$ datetime.now(timezone.utc).isoformat() + "Z"
> '2025-08-06T16:37:32.779313+00:00Z'
```

If you really wanna to ends with "Z" consider replacing instead
`datetime.now(timezone.utc).isoformat().replace('+00:00', 'Z')`

```
datetime.now(timezone.utc).isoformat().replace('+00:00', 'Z')
'2025-08-06T16:38:52.297460Z'
```